### PR TITLE
API/CLN: Add get_images; refactor Images; remove SubtractedImages

### PR DESCRIPTION
**Pull request by danielballan**
- `get_images` is a factory for `Images`, for consistency with get_\* funcs
- `SubtractedImages` should be done using `pims.pipeline`, and removing it
  removes dataportal's incongruous dependency on scikit-image
- the `Images` implementation is now faster, using get_events to read in
  only the events with the relevant data key

@sameera2004 Would you test this and then update the example notebook we wrote to use `get_images()` instead of `Images()`?

### CODE REVIEW

**Comment by sameera2004** [comment]
@danielballan sure

File: conda-recipe/meta.yaml [hunk 1]
```diff
@@ -20,7 +20,6 @@ requirements:
   run:
     - python
     - pandas
-    - matplotlib
     - pyyaml
     - pims
     - metadatastore >=0.1.0
```
Comment (ericdill) [inline conda-recipe/meta.yaml]: this should be changed to >= 0.2.0

Comment (ericdill) [inline conda-recipe/meta.yaml]: as should filestore

Comment (danielballan) [inline conda-recipe/meta.yaml]: both done

**Comment by ericdill** [comment]
I would be :+1: on including functionality that would introspect the header/descriptors and return one pims object per external data key.  Is that out of the scope of this PR?

**Comment by danielballan** [comment]
Yeah, out of scope IMO. This is (1) a hot-fix for v0.2.0 because `Images` issues warning messages (oops) and (2) a quick API consistency thing before I start documenting the new API.

I want pims v0.3 tagged before we rely on it any more than we already do. I'm skyping with Casper later today to make a final checklist.

**Comment by ericdill** [comment]
Alright, sounds good to me.  Did this need to be tested at CHX? Or is this good to go

**Comment by danielballan** [comment]
@sameera2004 I want to get this merged now because a couple forthcoming PRs will use it. I tested it on the notebook myself; it works as expected. When you change `Images` -> `get_images` it should work for you too. Let me know if you find any issues.

**Comment by ericdill** [comment]
LGTM

**Pull request merged**